src/gui/editors/AutomationEditor.cpp

```diff
@@ -728,353 +750,145 @@ void AutomationEditor::removePoints( int x0, int x1 )
 
 void AutomationEditor::mouseMoveEvent(QMouseEvent * mouseEvent )
 {
-	QMutexLocker m( &m_patternMutex );
+	QMutexLocker m(&m_patternEditorMutex);
 	if( !validPattern() )
 	{
 		update();
 		return;
 	}
 
-	if( mouseEvent->y() > TOP_MARGIN )
+	// If the mouse y position is inside the Automation Editor viewport
+	if (mouseEvent->y() > TOP_MARGIN)
 	{
-		float level = getLevel( mouseEvent->y() );
-		int x = mouseEvent->x();
+		float level = getLevel(mouseEvent->y());
+		// Get the viewport X position where the mouse is at
+		int x = mouseEvent->x() - VALUES_WIDTH;
 
-		x -= VALUES_WIDTH;
-		if( m_action == MOVE_VALUE )
-		{
-			x -= m_moveXOffset;
-		}
+		// Get the X position in ticks
+		int posTicks = (x * MidiTime::ticksPerBar()/m_ppb) + m_currentPosition;
 
-		int pos_ticks = x * MidiTime::ticksPerBar() / m_ppb +
-							m_currentPosition;
-		// m_mouseDownLeft used to prevent drag when drawing line
-		if (m_mouseDownLeft && m_editMode == DRAW)
+		switch(m_editMode)
 		{
-			if( m_action == MOVE_VALUE )
+			case DRAW:
 			{
-				// moving value
-				if( pos_ticks < 0 )
+				// We are dragging a node
+				if (m_mouseDownLeft)
 				{
-					pos_ticks = 0;
-				}
-
-				drawLine( m_drawLastTick, m_drawLastLevel,
-							pos_ticks, level );
-
-				m_drawLastTick = pos_ticks;
-				m_drawLastLevel = level;
-
-				// we moved the value so the value has to be
-				// moved properly according to new starting-
-				// time in the time map of pattern
-				m_pattern->setDragValue( MidiTime( pos_ticks ),
-								level, true,
-							mouseEvent->modifiers() &
-								Qt::ControlModifier );
-			}
+					if (m_action == MOVE_VALUE)
+					{
+						// When we clicked the node, we might have clicked slightly off
+						// so we account for that offset for a smooth drag
+						x -= m_moveXOffset;
+						posTicks = (x * MidiTime::ticksPerBar() / m_ppb) + m_currentPosition;
```

Comment:
I just noticed this isn't casting like the `alignedX` way up on line 508. Should we leave this doing integer division or do the `int(float(...))` thing like above?

Reply:
Previously `pos_ticks` had no casting and `align_x` did. I don't think they are necessary on neither: I think the alignment calculation could go without the castings (in the end we are making a float division and truncating it back to an integer), I just left it like that because that's how it was originally calculated. Leaving a TODO comment for now ("double check need for casting").

Reply:
Removed the casts from the `alignedX` variable, nice catch by the way!